### ISSUE TITLE
Fix bug caused by changing Metabase Analytics permissions

### ISSUE THREAD

**Issue by ellingtonjp** [flovouin/terraform-provider-metabase#48]
## Problem

The `metabase_permissions_graph` resource will error if the Metabase Analytics (pro feature) permissions are changed. Example:
```
╷
│ Error: Unexpected error while calling the Metabase API for operation 'get permissions graph'.
│ 
│ json: cannot unmarshal object into Go struct field PermissionsGraphDatabaseAccess.groups.data.schemas of type metabase.PermissionsGraphDatabaseAccessSchemas
╵
```

This is caused by Metabase changing the response to `/api/permissions/graph` to a schema different than what's in `metabase-api.yml`. Thus the client is erroring out.

## Reproducing

* Start a fresh instance of Metabase Pro using Postgres
  * The use of Postgres is important, for some reason I couldn't reproduce this with H2. Haven't tested any others.
* Go to Admin->Permissions->Collections->Metabase analytics
* Give "All Users" view permission

Apply the following terraform:
```
terraform {
  required_version = ">= 1.0"
  required_providers {
    metabase = {
      source  = "local/providers/metabase"
      version = "1.0.0"
    }
  }
}

provider "metabase" {
  endpoint = "http://localhost:3000/api"
  username = < user you created >
  password = < password you created >
}

import {
  to = metabase_permissions_graph.read
  id = 1
}

resource "metabase_permissions_graph" "read" {
  advanced_permissions = false
  permissions          = []
  provider = metabase
}
```

## Cause

After changing the metabase analytics permissions, response schema from `/api/permissions/graph` will differ from what's defined in `metabase-api.yml`, leading to errors in the generated client.

Here is a successful response. This was taken from a fresh instance of Metabase Pro with nothing but the sample database. Note the database `13371337` -- this is the metabase analytics database.
```
{
    "revision": 17,
    "groups": {
        "1": {
            "1": {
                "download": {
                    "native": "full",
                    "schemas": "full"
                }
            }
        },
        "2": {
            "1": {
                "data": {
                    "native": "write",
                    "schemas": "all"
                },
                "download": {
                    "native": "full",
                    "schemas": "full"
                },
                "data-model": {
                    "schemas": "all"
                },
                "details": "yes"
            },
            "13371337": {
                "data": {
                    "native": "write",
                    "schemas": "all"
                },
                "download": {
                    "native": "full",
                    "schemas": "full"
                },
                "data-model": {
                    "schemas": "all"
                },
                "details": "yes"
            }
        }
    },
    "sandboxes": [],
    "impersonations": []
}
```

Response after modifying permissions. Note how `schemas` has changed from a string to an object. The OpenAPI spec only allows strings: 
https://github.com/flovouin/terraform-provider-metabase/blob/607ca76bf1d4ab417a97a340aa45d97bec4c9bb6/metabase-api.yaml#L1282
```
{
    "revision": 17,
    "groups": {
        "1": {
            "1": {
                "download": {
                    "native": "full",
                    "schemas": "full"
                }
            },
            "13371337": {
                "data": {

//  PROBLEM: `schemas` is not a string:

                    "schemas": {
                        "public": {
                            "9": {
                                "query": "all"
                            },
                            "10": {
                                "query": "all"
                            },
                            "11": {
                                "query": "all"
                            },
                            "12": {
                                "query": "all"
                            },
                            "13": {
                                "query": "all"
                            },
                            "14": {
                                "query": "all"
                            },
                            "15": {
                                "query": "all"
                            },
                            "16": {
                                "query": "all"
                            },
                            "17": {
                                "query": "all"
                            },
                            "18": {
                                "query": "all"
                            },
                            "19": {
                                "query": "all"
                            },
                            "20": {
                                "query": "all"
                            },
                            "21": {
                                "query": "all"
                            }
                        }
                    }
                }
            }
        },
        "2": {
            "1": {
                "data": {
                    "native": "write",
                    "schemas": "all"
                },
                "download": {
                    "native": "full",
                    "schemas": "full"
                },
                "data-model": {
                    "schemas": "all"
                },
                "details": "yes"
            },
            "13371337": {
                "data": {
                    "native": "write",
                    "schemas": "all"
                },
                "download": {
                    "native": "full",
                    "schemas": "full"
                },
                "data-model": {
                    "schemas": "all"
                },
                "details": "yes"
            }
        }
    }
}
```

**Comment by ellingtonjp** [comment]
@flovouin I appreciate your work on this provider! It's been really helpful for us.

Here's a possible solution: https://github.com/flovouin/terraform-provider-metabase/pull/49

**Comment by flovouin** [comment]
Thanks for the detailed report!
I've just released version `0.5.1` with your fix. Hopefully it solves the issue for you.
Pro feature or not, I'm currently lacking some tests on permissions-related resources. I'll try to add them sooner than later.

**Comment by ellingtonjp** [comment]
@flovouin thanks for the fast response. We've tested the change in our environment and everything is working great.

Also apologies for not adding tests. I normally wouldn't submit a PR without tests but since there weren't already tests in place it felt a little out of my wheelhouse for now since I haven't worked on a tf provider before. If I get time to learn more I'll see if I can help out with them.

Thanks again for creating this provider!!
